### PR TITLE
Porting client from fern-reporter to client repo

### PR DESCRIPTION
**Pull request by mailprak**
This PR copies the client files from the fern-reporter repo to the fern-ginko-client repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Introduction
+
+Welcome to the Fern Project, an innovative open-source solution designed to enhance Ginkgo test reports. This project is focused on capturing, storing, and analyzing test data to provide insights into test performance and trends. The Fern Project is ideal for teams using Ginkgo, a popular BDD-style Go testing framework, offering a comprehensive overview of test executions and performance metrics.
+
+### Integrating the Client into Ginkgo Test Suites
+
+1. **Add the Fern dependency to your test project**:
+
+   ```bash
+   go get -u github.com/guidewire/fern-reporter
```
[the remark this answers is not in the document]
Comment (mailprak) [reply README.md]: Updated

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module client
+
+go 1.21.3
+
+require (
+	github.com/guidewire/fern-reporter v0.0.0-20240202054108-9e68ab4107a7
```
[the remark this answers is not in the document]
Comment (mailprak) [reply go.mod]: Removed the dependency with fern reporter.

File: client/go.mod [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module github.com/guidewire/fern-ginkgo-client/client
+
+go 1.21.3
+
+require (
+	github.com/guidewire/fern-reporter v0.0.0-20240202054108-9e68ab4107a7
```
[the remark this answers is not in the document]
Comment (mailprak) [reply client/go.mod]: The dependency with fern-reporter is removed.

**Comment by mailprak** [comment]
@anoop2811 The comments are addressed. Please review.

File: tests/adder_test.go
Comment (anoop2811) [inline tests/adder_test.go:1]: Good job. We will need to test this acceptance test in github actions. Perhaps you might want to add an issue wrt to that for this repo

**Review by anoop2811 — APPROVED** [review]
Can you create an issue in this repo for a github actions workflow which would run the test as well @mailprak ?

File: tests/adder_test.go
Comment (mailprak) [reply tests/adder_test.go:1]: done. Created an issue https://github.com/Guidewire/fern-ginkgo-client/issues/2

**Pull request merged**